x-pack/plugins/fleet/public/components/agent_enrollment_flyout/use_get_agent_incoming_data.tsx

```diff
@@ -92,8 +92,16 @@ export const usePollingIncomingData = (agentsIds: string[]) => {
         const { data } = await sendGetAgentIncomingData({ agentsIds });
 
         if (data?.items) {
-          setIncomingData(data?.items);
-          setIsLoading(false);
+          // filter out agents that have `data = false` and keep polling
+          const filtered = data?.items.filter((item) => {
+            const key = Object.keys(item)[0];
```

Comment:
in case of multiple `agentIds` being queried, do they come in separate objects in `items` array? so we are not missing anything by taking the first key only, right?

Reply:
No there's no problem there, the schema is something like
`[
    {
       'agentid1': {
           data: true
    },
   {
       'agentid2': {
           data: false
    },
    ...
]`